python/cuml/benchmark/algorithms.py

```diff
@@ -97,10 +100,14 @@ def run_cpu(self, data, **override_args):
         """Runs the cpu-based algorithm's fit method on specified data"""
         if self.cpu_class is None:
             raise ValueError("No CPU implementation for %s" % self.name)
+
         all_args = {**self.shared_args, **self.cpu_args}
         all_args = {**all_args, **override_args}
 
-        cpu_obj = self.cpu_class(**all_args)
+        if "cpu_obj" not in all_args:
```

Comment:
what does cpu_obj represent here? Seems like it is sometimes the results of the constructor of cpu_class and sometimes the data from setup_cpu_func?
Renaming and trimming down to just one use could help... maybe "extra_cpu_input" and only have it hold the results of the setup func? Or maybe I'm just misunderstanding...

Reply:
I renamed the output from setup functions to be `cpu_setup_result` and `cuml_setup_result`